src/preparers.py

```diff
@@ -533,68 +656,97 @@ def prepare_ultrachat(row):
     parent = "ultrachat"
     messages = []
     for i, script in enumerate(row["data"]):
-        messages.append({
-            "from": "user" if i % 2 == 0 else "assistant",
-            "text": script.strip(),
-            "parent": parent,
-        })
+        messages.append(
+            {
+                "from": "user" if i % 2 == 0 else "assistant",
+                "text": script.strip(),
+                "parent": parent,
+            }
+        )
+        parent = i
+    return messages
+
+
+def prepare_ultrachat_200k(row):
+    parent = "ultrachat_200k"
```

Comment:
Could we factorize the function that is the same between `prepare_ultrachat` and `prepare_ultrachat_200k` so they call the same sub-function with the different argument? (Since they seem identical otherwise.)

Reply:
Made fixes to merge the data prep functions. Using labels added during the download process.